src/Microsoft.AspNet.Http/RequestCookieCollection.cs

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Http.Internal
+{
+    public class RequestCookieCollection : IRequestCookieCollection
+    {
+        public static readonly RequestCookieCollection Empty = new RequestCookieCollection();
+#if DNXCORE50
+        private static readonly string[] EmptyKeys = Array.Empty<string>();
+        private static readonly StringValues[] EmptyValues = Array.Empty<string>();
+#else
+        private static readonly string[] EmptyKeys = new string[0];
+        private static readonly string[] EmptyValues = new string[0];
+#endif 
+        private static readonly Enumerator EmptyEnumerator = new Enumerator();
+        // Pre-box
+        private static readonly IEnumerator<KeyValuePair<string, StringValues>> EmptyIEnumeratorType = EmptyEnumerator;
+        private static readonly IEnumerator EmptyIEnumerator = EmptyEnumerator;
+
+        private Dictionary<string, string> Store { get; set; }
+
+        public RequestCookieCollection()
+        {
+        }
+
+        public RequestCookieCollection(Dictionary<string, string> store)
+        {
+            Store = store;
+        }
+
+        public RequestCookieCollection(int capacity)
+        {
+            Store = new Dictionary<string, string>(capacity, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public string this[string key]
+        {
+            get
+            {
+                if (key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                if (Store == null)
+                {
+                    return string.Empty;
+                }
+
+                string value;
+                if (TryGetValue(key, out value))
+                {
+                    return value;
+                }
+                return string.Empty;
+            }
+        }
+
+        StringValues IRequestCookieCollection.this[string key]
```

Comment:
Ah, right, this collection really is different from the others. It doesn't need to use StringValues anywhere, it only had that because of the old common IReadableStringCollection. We can just make this all `<string, string>` now.